### PR TITLE
New docs

### PR DESCRIPTION
**Pull request by Shqrdx**
Changes:
 Docs & tests for
-DepthFormat - enum values changed according to XNA
-EffectParameterClass
-EffectParameterType
-missed GraphicsProfile description

### CODE REVIEW

**Comment by Shqrdx** [comment]
@tomspilman Your thoughts ?

**Comment by tomspilman** [comment]
Looks good... merging!

File: Test/Framework/EnumConformingTest.cs [hunk 1]
```diff
@@ -9,6 +9,40 @@ namespace MonoGame.Tests.Framework
     class EnumConformingTest
     {
         [Test]
+        public void DepthFormatEnum()
+        {
+            Assert.AreEqual(0, (int)DepthFormat.None);
+            Assert.AreEqual(1, (int)DepthFormat.Depth16);
+            Assert.AreEqual(2, (int)DepthFormat.Depth24);
+            Assert.AreEqual(3, (int)DepthFormat.Depth24Stencil8);
```
Comment (tomspilman) [inline Test/Framework/EnumConformingTest.cs]: Did you run this test on XNA to be sure it was valid?

Comment (Shqrdx) [inline Test/Framework/EnumConformingTest.cs]: Yes of course.

Comment (Shqrdx) [inline Test/Framework/EnumConformingTest.cs]: And without these changes this test will not work in XNA.

File: MonoGame.Framework/Graphics/States/DepthFormat.cs [hunk 2]
```diff
@@ -1,54 +1,29 @@
...
-	// change to conform to XNA 4.0
+	/// <summary>
+	/// Defines formats for depth-stencil buffer.
+	/// </summary>
     public enum DepthFormat
     {
-		None = -1,
-		Depth16 = 54,
-		Depth24 = 51,
-		Depth24Stencil8 = 48,
```
Comment (Shqrdx) [inline MonoGame.Framework/Graphics/States/DepthFormat.cs]: And without these changes this test will not work in XNA

**Comment by tomspilman** [comment]
Ok... merging!

**Pull request merged**